test/jdk/java/awt/image/DrawImage/TiledImage.java

```diff
@@ -0,0 +1,225 @@
+import java.awt.image.DataBufferByte;
+import java.awt.image.RenderedImage;
+import java.awt.image.SampleModel;
+import java.awt.image.WritableRaster;
+import java.util.Objects;
+import java.util.Vector;
+
+
+/**
+ * @test
+ * @bug 8275345
+ * @summary RasterFormatException when drawing a tiled image made of non-writable rasters.
+ *
+ * Test drawing a tiled image made of non-writable {@link Raster} tiles.
+ * Drawing works when tiles are instances of {@link WritableRaster}.
+ * But if tiles are instances of {@link Raster} only, then the following
+ * exception is thrown:
+ *
+ * Exception in thread "main" java.awt.image.RasterFormatException: (parentX + width) is outside raster
+ *     at java.desktop/java.awt.image.WritableRaster.createWritableChild(WritableRaster.java:228)
+ *     at java.desktop/sun.java2d.SunGraphics2D.drawTranslatedRenderedImage(SunGraphics2D.java:2852)
+ *     at java.desktop/sun.java2d.SunGraphics2D.drawRenderedImage(SunGraphics2D.java:2711)
+ *
+ * The bug is demonstrated by drawing the same image twice:
+ * once with {@link WritableRaster} tiles (which succeed),
+ * then the same operation but with {@link Raster} tiles.
+ *
+ * The bug is caused by the following code in {@code SunGraphics2D}:
+ *
+ * // Create a WritableRaster containing the tile
+ * WritableRaster wRaster = null;
+ * if (raster instanceof WritableRaster) {
+ *     wRaster = (WritableRaster)raster;
+ * } else {
+ *     // Create a WritableRaster in the same coordinate system
+ *     // as the original raster.
+ *     wRaster =
+ *         Raster.createWritableRaster(raster.getSampleModel(),
+ *                                     raster.getDataBuffer(),
+ *                                     null);
+ * }
+ * // Translate wRaster to start at (0, 0) and to contain
+ * // only the relevant portion of the tile
+ * wRaster = wRaster.createWritableChild(tileRect.x, tileRect.y,
+ *                                       tileRect.width,
+ *                                       tileRect.height,
+ *                                       0, 0,
+ *                                       null);
+ *
+ * If {@code raster} is not an instance of {@link WritableRaster},
+ * then a new {@link WritableRaster} is created wrapping the same
+ * buffer <strong>but with a location of (0,0)</strong>, because
+ * the {@code location} argument of {@code createWritableRaster}
+ * is null. Consequently the call to {@code createWritableChild}
+ * shall not be done in that case, because the raster is already
+ * translated. The current code applies translation twice.
+ *
+ * This bug is largely unnoticed because most {@code Raster.create}
+ * methods actually create {@link WritableRaster} instances, even
+ * when the user did not asked for writable raster. To make this
+ * bug apparent, we need to invoke {@code Raster.createRaster(…)}
+ * with a sample model for which no optimization is provided.
+ */
+final class TiledImage implements RenderedImage {
+    /**
```

Comment:
Still not right. Now you added the tags I ran under jtreg and it errored out.
result: Error. java.lang.IllegalAccessException: class com.sun.javatest.regtest.agent.MainWrapper$MainThread cannot access a member of class TiledImage with modifiers "public static

Basically change "final" to "public" - or at least add public. 

Might be prudent to try this for yourself - and useful for the future.
jtreg instructions for building it can be found here : https://openjdk.java.net/jtreg/build.html
Very possibly there's someone that provides a pre-built binary. There's none on OpenJDK.

Reply:
Thanks for the tips and your patience. Actually I tried to run jtreg yesterday before to commit the change, but got the following error (I wanted to try on Java 17 first in order to verify that the test failed):

```shell
jtreg -jdk:$JAVA_HOME test/jdk/java/awt/image/DrawImage/TiledImage.java
```
```
failed to get value for vm.cds.write.archived.java.heap
java.lang.UnsatisfiedLinkError: 'boolean jdk.test.whitebox.WhiteBox.canWriteJavaHeapArchive()'
	at jdk.test.whitebox.WhiteBox.canWriteJavaHeapArchive(Native Method)
	at requires.VMProps.vmCDSCanWriteArchivedJavaHeap(VMProps.java:413)
	at requires.VMProps$SafeMap.put(VMProps.java:72)
	at requires.VMProps.call(VMProps.java:113)
	at requires.VMProps.call(VMProps.java:60)
	at com.sun.javatest.regtest.agent.GetJDKProperties.run(GetJDKProperties.java:80)
	at com.sun.javatest.regtest.agent.GetJDKProperties.main(GetJDKProperties.java:54)
```
I'm sure I did something wrong with my configuration…

Reply:
Alas, I'm purely a user as far as jtreg goes. I've had odd looking errors myself and in those cases the eventual answer was often that I was running the wrong (too old) version of jtreg.
I hunt jtreg down from an internal build archive so I can't point you at what I use (nor provide it) but all the external contributors must have some favoured place to get their copy. I just don't know what it is. the code-tools list might be able to help.

But if you want to expedite this, just make the one change - I already tested it.

Reply:
@desruisseaux You can take the jtreg builds from @shipilev: https://builds.shipilev.net/jtreg/